src/main/kotlin/it/pagopa/ecommerce/payment/requests/client/NodoPerPmClient.kt

```diff
@@ -28,16 +28,16 @@ public class NodoPerPmClient(
       .body(Mono.just(request), CheckPositionDto::class.java)
       .retrieve()
       .onStatus(HttpStatus::isError) { clientResponse ->
-        clientResponse.bodyToMono(String::class.java).flatMap { errorResponseBody: String ->
-          Mono.error(ResponseStatusException(HttpStatus.BAD_GATEWAY, errorResponseBody))
-        }
+        Mono.error(
+          RestApiException(
+            clientResponse.statusCode(),
```

Comment:
In this way we are not remapping Nodo errors but echoing the received http status code, maybe a remapping is needed here.
Ex: Nodo 500 --> payment request 502
What do you think?

Reply:
I catch a httpstatus isError that includes 4XX and 5XX response code are you telling me that we turn them into 502 ? Or do we apply different logics to each node response code ?